var/spack/repos/builtin/packages/damask-grid/package.py

```diff
@@ -17,9 +17,10 @@ class DamaskGrid(CMakePackage):
 
     depends_on('pkgconfig', type='build')
     depends_on('cmake@3.10:', type='build')
+    depends_on('petsc+mpi+hdf5')
     depends_on('petsc@3.14.0:3.14,3.15.1:3.15', when='@3.0.0-alpha4')
     depends_on('petsc@3.14.0:3.14,3.15.1:3.16', when='@3.0.0-alpha5')
-    depends_on('hdf5+fortran')
+    depends_on('hdf5@1.12:+mpi+fortran')
```

Comment:
Don't older versions of damask work with older versions of hdf5? If the new hdf5 version and hdf5 requirement are for a new version of damask you can leave those lines in place and add:
```python
    depends_on('petsc+mpi+hdf5', when='@xxxx:')
    depends_on('hdf5@1.12+mpi', when='@xxxx:')
```

Reply:
Thanks for the hint. Even newer versions of DAMASK compile with HDF5 1.10, but we have ongoing issues with MPI-output which are partly resolved with 1.12. Parallel IO is therefore disabled for versions < 1.12. Using 1.10 should be considered as a fallback option.
From the discussion in the other merge request, I remember that version requirements should be hard requirements, not recommendations. After https://github.com/spack/spack/commit/dce4f161d4cebf95cd604cc1fbca284ea0e9c622, I therefore change the requirement to 1.10 (1.8 was never tested) and assume that most people will have a working default setup.